tests/integrated/test-laplace/test_laplace.cxx

```diff
@@ -12,74 +12,95 @@ int main(int argc, char **argv) {
   // Initialise BOUT++, setting up mesh
   BoutInitialise(argc, argv);
 
-  FieldFactory f(mesh);
+  FieldFactory f{mesh};
 
   Field3D input = f.create3D("(1-gauss(x-0.5,0.2))*gauss(y-pi)*gauss(z-pi)");
   Field2D a = f.create2D("gauss(x) * sin(y)");
   Field2D c = f.create2D("sin(x) * gauss(x-0.5) * gauss(y-pi)");
   Field2D d = f.create2D("y - pi/2");
   SAVE_ONCE4(input, a, c, d);
 
-  Field3D flag0 = invert_laplace(input, 0);
-  Field3D flag3 = invert_laplace(input, 3);
+  auto lap = std::unique_ptr<Laplacian>{Laplacian::create()};
+
+  lap->setCoefA(0.0);
+  lap->setCoefC(1.0);
+  lap->setCoefD(1.0);
+  Field3D flag0 = lap->solve(input);
+  lap->setInnerBoundaryFlags(INVERT_DC_GRAD + INVERT_AC_GRAD);
```

Comment:
It's a bit worrying that we don't test the outer-boundary-condition flags here. It'd be nice to at least set both inner- and outer-flags to the same thing. Then we'd have to update the benchmark data as well. Maybe there's a way to hack around that though... `input` and `a` are even in `(x-0.5)`, while `c` is odd, so if we duplicate each test, setting the outer boundary flags instead of the inner boundary flags then (remembering the `c` term is `1/C * Grad_perp(C)*Grad_perp(f)`, so it's sign doesn't flip even if `c`'s does) we'd have a new solution `f2(x-0.5,z)=f(-(x-0.5),z)`, where `f(x,z)` is the solution with the inner boundary condition set. So in `runtest` I think we'd just have to reverse the x-dimension of the stored solution (for the inner-boundary set case) to get the expected solution with outer boundary flags set.

Reply:
Are you ok if we hold off on fixing this now, and include it in whatever rewrite of the Laplacians we end up doing?

Reply:
yep, makes sense to me.